integration/test_timescaledb.py

```diff
@@ -306,3 +303,356 @@ async def test_timescaledb_async_emit(timescaledb):
     cursor.close()
 
     assert count == 2, "Async emission should insert exactly 2 rows"
+
+
+def test_timescaledb_schema_validation_missing_required_column(timescaledb):
+    """Test that validation properly catches missing required (non-nullable) columns"""
+    (connection, _, _, _, _, _, table_name, dsn_url, _, _) = timescaledb
+
+    # Create a table with a non-nullable column
+    cursor = connection.cursor()
+    validation_table = f"{table_name}_validation"
+    with contextlib.suppress(PsycopgError):
+        cursor.execute(f"DROP TABLE IF EXISTS {validation_table};")
+
+    cursor.execute(
+        f"""
+        CREATE TABLE {validation_table} (
+            time TIMESTAMPTZ NOT NULL,
+            required_col INTEGER NOT NULL,
+            optional_col VARCHAR(50)
+        );
+    """
+    )
+    cursor.execute(f"SELECT create_hypertable('{validation_table}', 'time');")
+    cursor.close()
+
+    time_format = "%d/%m/%y %H:%M:%S UTC%z"
+    controller = build_flow(
+        [
+            SyncEmitSource(),
+            TimescaleDBTarget(
+                dsn=dsn_url,
+                table=validation_table,
+                time_col="time",
+                columns=["required_col", "optional_col"],
+                time_format=time_format,
+                max_events=10,
+            ),
+        ]
+    ).run()
+
+    # Test data missing the required column
+    invalid_data = {
+        "time": "18/09/19 01:55:10 UTC+0000",
+        "optional_col": "test_value",
+        # Missing required_col (non-nullable)
+    }
+
+    # Should raise ValueError for missing required column
+    with pytest.raises(ValueError, match=r"Missing required non-nullable column 'required_col'"):
+        controller.emit(invalid_data, None)
```

Comment:
No reason to pass `None`, especially by position.
```suggestion
        controller.emit(invalid_data)
```

Reply:
Applies elsewhere as well.

Reply:
will fix